src/parse_tree.rs

```diff
@@ -98,27 +98,45 @@ impl Root {
 			match c {
 				c if c.is_whitespace() => {}
 				c if c.is_digit(10) => {
-					todo!("Ziffer in Zahl konvertieren und auf den Stapel legen")
+					if let Some(digit) = c.to_digit(10) {
+						expr_stack.push(Expr::Int(digit as i64));
+					} 
 				}
 				c if c.is_ascii_lowercase() => {
-					todo!("Variablenname auf den Stapel legen")
+					expr_stack.push(Expr::Var(c));
 				}
+
 				'+' => {
-					todo!("Additionsknoten auf den Stapel legen")
+					if let (Some(right), Some(left)) = (expr_stack.pop(), expr_stack.pop()) {
+						expr_stack.push(Expr::Add(Box::new(left), Box::new(right)));
+					}
 				}
 				'-' => {
-					todo!("Subtraktionsknoten auf den Stapel legen")
+					if let (Some(right), Some(left)) = (expr_stack.pop(), expr_stack.pop()) {
+						expr_stack.push(Expr::Sub(Box::new(left), Box::new(right)));
+					}
 				}
 				'*' => {
-					todo!("Multiplikationsknoten auf den Stapel legen")
+					if let (Some(right), Some(left)) = (expr_stack.pop(), expr_stack.pop()) {
+						expr_stack.push(Expr::Mul(Box::new(left), Box::new(right)));
+					}
 				}
 				'/' => {
-					todo!("Divisionsknoten auf den Stapel legen")
+					if let (Some(right), Some(left)) = (expr_stack.pop(), expr_stack.pop()) {
+						expr_stack.push(Expr::Div(Box::new(left), Box::new(right)));
+					}
```

Comment:
Refactor example: 
```rust
'+' | '-' | '*' | '/' => {
                    // first popped is right to assert correct non-commutative operations
                    if let (Some(rhs), Some(lhs)) = (expr_stack.pop(), expr_stack.pop()) {
                        let operation = match token {
                            '+' => Expr::Add(Box::new(lhs), Box::new(rhs)),
                            '-' => Expr::Sub(Box::new(lhs), Box::new(rhs)),
                            '*' => Expr::Mul(Box::new(lhs), Box::new(rhs)),
                            '/' => Expr::Div(Box::new(lhs), Box::new(rhs)),
                            _ => unreachable!(), // Shouldn't happen
                        };
                        expr_stack.push(operation)
                    } 
```

Reply:
Done. I thought about doing this but i was way to fixated at rusts syntax that i didnt even bother

Reply:
The thing is how would i be able to do this, normally in this case i would check both lhs and rhs with maybe an if statement to check if they are both integers so something like: if rhs = Expr::Int(_) && lhs = Expr::Int(_){ ... }, or something like that but this doesnt work for many reasons one of them the usage of (_) but other than that i dont understand why this doesnt work.